### ISSUE TITLE
Problem with gcc-9.2.0-r3(gentoo) -Werror compiler flag. -Werror=address-of-packed-member

### ISSUE THREAD

**Issue by D-Y-V** [aerospike/aerospike-server#14]
When we are use simple make (assume -Wall compiler flag)
$ make 
<-----------snip-------------->
server/ai/src»
cc -gdwarf-4 -g3 -O3  -fno-common -fno-strict-aliasing -Wall -D_FILE_OFFSET_BITS=64 -std=gnu99 -D_REENTRANT -D_GNU_SOURCE -DENHANCED_ALLOC  -march=nocona -MMD -Werror -DMARCH_x86_64 -D__FILENAME__=\"ai_btree.c\" -o ../../target/Linux-x86_64/obj/ai_btree.o -c -I/home/user_name/aero/server/modules/jemalloc/include -I../include -I../../target/Linux-x86_64/gen -I/home/user_name/aero/server/cf/include -I/home/user_name/aero/server/as/include -I/home/user_name/aero/server/as/include -I/home/user_name/aero/server/modules/common/target/Linux-x86_64/include -I/home/user_name/aero/server/modules/mod-lua/target/Linux-x86_64/include ai_btree.c
In file included from /home/user_name/aero/server/as/include/base/secondary_index.h:30,
                 from ../include/ai_types.h:37,
                 from ../include/ai_obj.h:30,
                 from ai_btree.c:29:
/home/user_name/aero/server/as/include/base/datamodel.h: In function «as_bin_set_id_from_name_w_len»:
/home/user_name/aero/server/as/include/base/datamodel.h:1293:67: ошибка: taking address of packed member of «struct as_bin_s» may result in an unaligned pointer value [-Werror=address-of-packed-member]
 1293 |  return as_bin_get_or_assign_id_w_len(ns, (const char *)buf, len, &b->id);
      |                                                                   ^~~~~~
cc1: all warnings interpreted as error
make[2]: *** [../../make_in/Makefile.targets:13: ../../target/Linux-x86_64/obj/ai_btree.o] Err 1
make[2]: leave «/home/user_name/aero/server/ai/src»
make[1]: *** [Makefile:9: all] Err 2
make[1]: leaving «/home/user_name/aero/server/ai»
make: *** [Makefile:53: all] Err 2

**Comment by karlhiramoto** [comment]
Same error also occurs with ubunu 20.04 with gcc 9.3

```
gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/9/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none:hsa
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 9.3.0-10ubuntu2' --with-bugurl=file:///usr/share/doc/gcc-9/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++,gm2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-9 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none,hsa --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 9.3.0 (Ubuntu 9.3.0-10ubuntu2) 
```

```
cc -gdwarf-2 -g3 -O3  -fno-common -fno-strict-aliasing -Wall -D_FILE_OFFSET_BITS=64 -std=gnu99 -D_REENTRANT -D_GNU_SOURCE -DENHANCED_ALLOC  -march=nocona -MMD -Werror -DMARCH_x86_64 -D__FILENAME__=\"ai_btree.c\" -o ../../target/Linux-x86_64/obj/ai_btree.o -c -I/home/karl/Work/aerospike-server/modules/jemalloc/include -I../include -I../../target/Linux-x86_64/gen -I/home/karl/Work/aerospike-server/cf/include -I/home/karl/Work/aerospike-server/as/include -I/home/karl/Work/aerospike-server/as/include -I/home/karl/Work/aerospike-server/modules/common/target/Linux-x86_64/include -I/home/karl/Work/aerospike-server/modules/mod-lua/target/Linux-x86_64/include ai_btree.c
In file included from /home/karl/Work/aerospike-server/as/include/base/secondary_index.h:30,
                 from ../include/ai_types.h:37,
                 from ../include/ai_obj.h:30,
                 from ai_btree.c:29:
/home/karl/Work/aerospike-server/as/include/base/datamodel.h: In function ‘as_bin_set_id_from_name_w_len’:
/home/karl/Work/aerospike-server/as/include/base/datamodel.h:1308:67: error: taking address of packed member of ‘struct as_bin_s’ may result in an unaligned pointer value [-Werror=address-of-packed-member]
 1308 |  return as_bin_get_or_assign_id_w_len(ns, (const char *)buf, len, &b->id);
      |                                                                   ^~~~~~
```

**Comment by kportertx** [comment]
A fix for this issue has been queued for 5.1, thanks @karlhiramoto.

**Comment by kportertx** [comment]
This has been resolved with the release of 5.1 source code.
